src/history_manager.py

```diff
@@ -40,29 +71,32 @@ def _delete_operation(self, op):
 	# Controls accessed by DrImage #############################################
 
 	def try_undo(self):
+	
 		if self._operation_is_ongoing():
 			self._image.active_tool().cancel_ongoing_operation()
 			return
-		if len(self._undo_history) > 0:
-			last_op = self._undo_history.pop()
-			self._redo_history.append(last_op)
+
+		undone_op = self._undo_history[-1].pop_last_operation()
+		if undone_op is not None:
+			# Dont save reloads from disk to redo history
```

Comment:
if there is a technical reason please explain it in the comment, otherwise you should try to have them in the redo history: if the image has been reloaded, it can be because it has been changed on the disk by another app, and we don't want to lose those changes if the user undoes operations

Reply:
if the reason is that it's a pain in the ass to support this special case, then write it anyway but with "TODO" in the comment

Reply:
I wrote a comment with a TODO label regarding this. With the current try_redo implementation, saving the reload to the redo history is causing the redo button to be always enabled even if there's nothing to redo. 

It can probably be fixed, but I'm a bit busy right now. I'll try to think about it in a couple weeks.

Reply:
don't worry it's not working with my former implementation either lol, i don't mind a few minor problems i just need comments to keep track of them